src/pages/CartPage.js

```diff
@@ -0,0 +1,60 @@
+import { useContext } from 'react';
+import { useHistory } from 'react-router-dom';
+import Card from '../components/common/Card';
+import { FaTrash } from "react-icons/fa";
+import StyledFlex from '../components/common/StyledFlex';
+import ShoppingCartContext from '../utils/state/ShoppingCartContext';
+import ItemCounter from '../components/common/ItemCounter';
+import { StyledIconWrapper } from '../components/common/StyledIconWrapper';
+const CartPage = () => {
+  const { state, dispatch } = useContext(ShoppingCartContext);
+  let history = useHistory();
+
+  if(state.items.length === 0) {
+    return <StyledFlex alignItems={'center'} justifyContent={'center'}><h1>Cart is empty</h1></StyledFlex>
+  }
+  const handleOnChange = (e, item) => {
+    if(item.data.stock - parseInt(e.target.value) < 0) return;
+    dispatch({
+      type: "UPDATE_ITEM",
+      payload: {
+        ...item, 
+        count: parseInt(e.target.value),
+        remCount: parseInt(e.target.value) - item.count
```

Comment:
This operation should be inside the reducer because if you have other places where you dispatch the same action you need to repeat this.